sql/src/main/java/io/crate/execution/ddl/tables/AlterTableOperation.java

```diff
@@ -404,7 +403,6 @@ static PutIndexTemplateRequest preparePutIndexTemplateRequest(IndexScopedSetting
         // merge settings
         Settings.Builder settingsBuilder = Settings.builder();
         settingsBuilder.put(indexTemplateMetaData.settings());
-        settingsBuilder.put(newSettings);
```

Comment:
I think the whole settings handling here can be removed, so no settings will be set on the `PutIndexTemplateRequest`

Reply:
The test itself must stay, but it must be refactored as well as the logic for removing private settings is now inside the new `AlterTableClusterStateExecutor`.